docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,221 @@
+increase in data being stored. An example being we now generate ~430,000
+`DigestRunSubscriber` records each day which is about eight times the amount
+compared to that of ten months ago.
+
+We've also gained more insight into what data we're not using for analytics.
+The original plan documented in the previous retention policy was to retain as
+much data as possible to enable us to do as much analysis as needed in the
+future. However, much of this data is never used for analysis and so doesn't
+serve a purpose in the email system.
+
+Contrary to what was previously thought, storing lots of unused data
+indefinitely actually complicates and pollutes our auditing and analysis. An
+example of this can be seen if we look at subscriber lists. We have over
+~73,000 lists with active subscribers but in the last week alone, we've
+accumulated 1139 lists which don't have any subscriptions at all (ended or
+active). These unused lists prevent us from doing doing simple auditing tasks
+easily, like determining which lists are actually subscribed to.
+
+Knowing the above, we thought about how we could cleanup this unused data to
+keep EmailAlertApi's database manageable going forward. There are a couple of
+different approaches to consider, including storing unused data in cloud
+storage away from the systems database (like we do when [archiving] emails) and
+subsequently deleting the data locally. Or, setting up a means of removing the
+data we don't use from the system entirely.
+
+[archiving]: ...
+[archived]: ...
+[deleted]: ...
+[nullified]: ...
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md
+[migrates subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+## Decision
+
+We decided that historic (unused) data should be removed periodically from the
+system entirely, instead of retaining it through archiving. Archiving was
+deemed as unnecessary because if we've deemed data as being historic then we
+don't see a use for it (analytical or otherwise) going forward and so retaining
+it via an external data store seems a waste of resources and energy in
+maintaining the process.
+
+As part of the decision we concluded that the data we, and the system no longer
+uses should only exist for a maximum of one year before removal due to there
+being no evidence of needing unused data for analytics beyond one year. There
+is however some bespoke behaviour for subscriber lists where we delete certain
+lists sooner than a year but that will be elaborated on in a later section.
+
+We also concluded that we don't need to change the retention for emails (and
+subscription contents) as we already clean them up via the [email deletion
+mechanism] which is sufficient.
+
+The following sections document our approach to the decision for different data
+in the system.
+
+### Data which has previously been used
```

Comment:
I'm finding the formatting in this section quite hard to follow, especially with the hrule, which cuts the whole document in half.

I appreciate there are quite a few levels at this point. Something which might work better would be to use a list, like this:

- **ContentChange, Message & DigestRun**. Remove once their created_at value is over a year old.

...

- **SubscriberList**. We remove lists when they have no active subscriptions and their latest, ended subscription is over a year old. It is possible for a list we deem as historic to be subscribed to again, however this seems unlikely due to the list not having any active subscriptions for over a year. If this was to happen after the list has been deleted then a new SubscriberList record will be generated.

Reply:
Arghglhjsncdp I did this originally. 😆 

I've used a list which does make the section easier to read.

Let me know if it's easier to follow! 👍